### PR TITLE
:bug: Fix SignTx: use SuggestGasPrice instead of EstimateGas for GasPrice

### PR DESCRIPTION
**Pull request by poteto0**
## Summary

- `wallet.SignTx` was assigning the return value of `EstimateGas` (gas **units**, e.g. 21000) to `txRequest.GasPrice` (wei **per** gas unit), sending transactions with effectively-zero gas price that are never mined — closes #331
- Replace the wrong assignment with a call to `SuggestGasPrice()` when `GasPrice` is nil and the request is not EIP-1559; caller-supplied `GasPrice` is left untouched
- Add 4 new test cases: nil GasPrice auto-filled from `SuggestGasPrice`, caller's GasPrice preserved, `SuggestGasPrice` error propagated, EIP-1559 tx skips `SuggestGasPrice`

## Gas concepts (why the old code was wrong)

| Field | Meaning | Unit |
|-------|---------|------|
| `EstimateGas` return | gas units the tx will consume | gas units (e.g. 21,000) |
| `GasLimit` | max gas units caller allows | gas units |
| `GasPrice` | fee per gas unit | wei / gas |

`GasLimit < estimatedGas` guard is **correct** (both are gas units).  
`txRequest.GasPrice = estimatedGas` was the **bug** (units ≠ price).

## Test plan

- [ ] `go test ./wallet/... -v -count=1` — all 14 `TestWallet_SignTx` subtests pass
- [ ] `go test ./wallet/... -race` — no races
- [ ] New tests exercise: nil GasPrice auto-fill, caller price preserved, SuggestGasPrice error, EIP-1559 bypass
- [ ] Pinned r/s signature values updated to match new 1-Gwei GasPrice

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: wallet/wallet.go [hunk 1]
```diff
@@ -234,7 +233,14 @@ func (w *wallet) SignTx(txRequest types.TransactionRequest) (*gethTypes.Transact
 			estimatedGas.Uint64(),
 		)
 	}
-	txRequest.GasPrice = estimatedGas
+
```
Comment (poteto0) [inline wallet/wallet.go:236]: ここにMaxFeePerGas==nil
と
MaxPriorityFeePerGas==nil
をチェックしている理由をコメント

**Review by poteto0 — COMMENTED** [review]
LGTM
